presto-pinot/src/main/java/io/prestosql/pinot/PinotColumnHandle.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.pinot;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.type.Type;
+
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class PinotColumnHandle
+        implements ColumnHandle
+{
+    private final String columnName;
+    private final Type dataType;
+    private final PinotColumnType type;
+
+    @JsonCreator
+    public PinotColumnHandle(
+            @JsonProperty("columnName") String columnName,
+            @JsonProperty("dataType") Type dataType,
+            @JsonProperty("type") PinotColumnType type)
+    {
+        this.columnName = requireNonNull(columnName, "column name is null");
+        this.dataType = requireNonNull(dataType, "data type name is null");
+        this.type = requireNonNull(type, "type is null");
+    }
+
+    @JsonProperty("columnName")
```

Comment:
`"columnName"` in the annotation is unnecessary as it matches the method name. Same for `"dataType"` below.